docs/configuring-metabase/config-file.md

```diff
@@ -0,0 +1,98 @@
+---
+title: "Metabase configuration file"
+---
+
+# Metabase configuration file
+
+{% include plans-blockquote.html feature="Loading from a configuration file" self-hosted-only="true" %}
+
+On some paid, self-hosted plans, Metabase supports initialization on launch from a config file named `config.yml`. This file should either be in the current directory (the directory where the running Metabase JAR is located), or in a path specified by the `MB_CONFIG_FILE_PATH` [environment variable](./environment-variables.md). In this config file, you can specify:
+
+- Settings
+- Users
+- Databases
+
+The settings as defined in the config file work the same as if you set these settings in the Admin Settings in your Metabase. Settings defined in this configuration file will update any existing settings. If, for example, a database already exists (that is, you'd already added it via initial set up or **Admin settings** > **Databases**, Metabase will update the database entry based on the data in the config file. 
+
+The config file settings are not treated as a hardcoded source of truth like [environment variables](./environment-variables.md) are.
+
+## Example `config.yml` file
+
+This file sets up a user account and two database connections.
+
+```
+version: 1
+config:
+  users:
+    - first_name: Cam
+      last_name: Era
+      password: 2cans3cans4cans
+      email: cam@example.com
+  databases:
+    - name: test-data (Postgres)
+      engine: postgres
+      details:
+        host: localhost
+        port: 5432
+        password: {{ env POSTGRES_TEST_DATA_PASSWORD }}
+        dbname: test-data
+    - name: Sample Dataset (Copy)
+      engine: h2
+      details:
+        db: "/home/cam/metabase/resources/sample-database.db;USER=GUEST;PASSWORD={{env SAMPLE_DATASET_PASSWORD}}"
+```
+
+To determine which keys you can specify, simply look at the fields available in Metabase itself.
+
+## Referring to environment variables in the `config.yml`
+
+Environment variables can be specified with `{% raw %}{{ template-tags }}{% endraw %}` like `{% raw %}{{ env POSTGRES_TEST_DATA_PASSWORD }}{% endraw %}` or `{% raw %}[[options {{template-tags}}]]{% endraw %}`.
+
+Metabase doesn't support recursive expansion, so if one of your environment variables references _another_ environment variable, you're going to have a bad time.
+
+## Disable initial database sync
+
+To disable the initial database sync, you can add `config-from-file-sync-database` to the `settings` list and set the value to `false`. The setting `config-from-file-sync-database` must come _before_ the databases list, like so:
```

Comment:
I think it is worth pointing out why users would want to do this.

This only makes sense when they expect to load the data model from a serialized instance and disable the scheduler.